chanfitness/chaneventstore_testctx_test.go

```diff
@@ -0,0 +1,338 @@
+	wire.OutPoint) {
+
+	vertex, pubKey, chanPoint := c.newChannel()
+	c.sendChannelOpenedUpdate(pubKey, chanPoint)
+
+	return vertex, pubKey, chanPoint
+}
+
+// closeChannel sends a close channel event to our subscribe server.
+func (c *chanEventStoreTestCtx) closeChannel(channel wire.OutPoint) {
+	update := channelnotifier.ClosedChannelEvent{
+		CloseSummary: &channeldb.ChannelCloseSummary{
+			ChanPoint: channel,
+		},
+	}
+
+	c.channelSubscription.sendUpdate(update)
+}
+
+// tickFlapCount forces a tick for our flap count ticker with the current time.
+func (c *chanEventStoreTestCtx) tickFlapCount() {
+	testTicker := c.store.cfg.FlapCountTicker.(*ticker.Force)
+
+	select {
+	case testTicker.Force <- c.store.cfg.Clock.Now():
+
+	case <-time.After(timeout):
+		c.t.Fatalf("could not tick flap count ticker")
+	}
+}
+
+// peerEvent sends a peer online or offline event to the store for the peer
+// provided.
+func (c *chanEventStoreTestCtx) peerEvent(peer route.Vertex, online bool) {
+	var update interface{}
+	if online {
+		update = peernotifier.PeerOnlineEvent{PubKey: peer}
+	} else {
+		update = peernotifier.PeerOfflineEvent{PubKey: peer}
+	}
+
+	c.peerSubscription.sendUpdate(update)
+}
+
+// sendChannelOpenedUpdate notifies the test event store that a channel has
+// been opened.
+func (c *chanEventStoreTestCtx) sendChannelOpenedUpdate(pubkey *btcec.PublicKey,
+	channel wire.OutPoint) {
+
+	update := channelnotifier.OpenChannelEvent{
+		Channel: &channeldb.OpenChannel{
+			FundingOutpoint: channel,
+			IdentityPub:     pubkey,
+		},
+	}
+
+	c.channelSubscription.sendUpdate(update)
+}
+
+// assertEvents asserts that a channel event store has an expected set of
+// events. This function can only be called once the test context's store has
+// been stopped.
+func (c *chanEventStoreTestCtx) assertEvents(channel wire.OutPoint,
+	events []channelEvent) {
+
+	c.requireStopped()
+
+	eventLog, ok := c.store.channels[channel]
+	require.True(c.t, ok, "channel not found")
+
+	// Check that our opened time is always non-zero, covering a
+	// previous bug where open times were not set for channels with
+	// peers who are offline when we add the channel to the store.
+	require.False(c.t, eventLog.openedAt.IsZero(), "zero open time")
+
+	// Assert that we have the number of events we expect.
+	actualEvents := eventLog.listEvents()
+	require.Len(c.t, actualEvents, len(events), "event count wrong")
+
+	// Run through our expected events and check that we have the
+	// right kind. We do not check time (at this stage) because it
+	// is not mocked.
+	for i, event := range events {
+		require.Equal(c.t, event.eventType, actualEvents[i].eventType,
+			"event type wrong")
+	}
+}
+
+// assertPeerInfo asserts that we have tracked our peers online state and uptime
+// as expected. This function can only be called once the test context's store
+// has been stopped.
+func (c *chanEventStoreTestCtx) assertPeerInfo(expected map[route.Vertex]peerInfo) {
```

Comment:
nit: line too long

Reply:
Thanks for pointing these out! In some cases, such as this one and a few `generateEvents` functions above, I've intentionally chosen to not line wrap. In the contributor guidelines, we make a note that wrapping at 80 should be done on a [best effort basis](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#48-additional-style-constraints-on-top-of-gofmt).

When we have a single long parameter for a function, I find it more readable to run over than add the newline (or make the var name less descriptive):
```
func (c *chanEventStoreTestCtx) assertPeerInfo(
    expected map[route.Vertex]peerInfo) {
```

This is a matter of personal taste, so there can always be arguments in either direction. Thankfully we have `gofmt` to save us time discussing these things in most cases!

Reply:
Cool thanks for letting me know!

Reply:
Thanks for reviewing! Always really helpful to have extra eyes :D